fix(transformer/typescript): namespaces disappeared and do not transform when a type-only namespace followed by a value module namespace

Dunqing opened the pull request:
Input:
```ts
// SymbolFlags: NameSpaceModule
export namespace Foo {
	 export type T = 0;
}
// SymbolFlags: ValueModule
export namespace Foo {
	 export const Bar = 1;
}
```

Before output:
```js
expot {}
```

After Output:
```js
// SymbolFlags: ValueModule
export let Foo;
(function(_Foo) {
  const Bar = _Foo.Bar = 1;
})(Foo || (Foo = {}));
```

When both `NameSpaceModule` and `ValueModule` are present, we need to check the current
declaration flags. If the current declaration is `NameSpaceModule`, we can return early
because it's a type-only namespace and doesn't emit any JS code, otherwise we need to
continue transforming it.

Dunqing commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10511?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#10511** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10511?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10511?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#10510** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10510?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=10511).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **Apr 21, 5:19 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
fix(transformer/typescript): namespaces disappeared and do not transform when a type-only namespace followed by a value module namespace (#10511)

Input:
```ts
// SymbolFlags: NameSpaceModule
export namespace Foo {
	 export type T = 0;
}
// SymbolFlags: ValueModule
export namespace Foo {
	 export const Bar = 1;
}
```

Before output:
```js
expot {}
```

After Output:
```js
// SymbolFlags: ValueModule
export let Foo;
(function(_Foo) {
  const Bar = _Foo.Bar = 1;
})(Foo || (Foo = {}));
```

When both `NameSpaceModule` and `ValueModule` are present, we need to check the current
declaration flags. If the current declaration is `NameSpaceModule`, we can return early
because it's a type-only namespace and doesn't emit any JS code, otherwise we need to
continue transforming it.
